src/controllers/scripting/legacy/scriptconnection.cpp

```diff
@@ -1,8 +1,12 @@
 #include "controllers/scripting/legacy/scriptconnection.h"
 
 #include "controllers/scripting/legacy/controllerscriptenginelegacy.h"
+#include "util/trace.h"
 
 void ScriptConnection::executeCallback(double value) const {
+    std::unique_ptr<Trace> pCallCallbackTrace;
+    pCallCallbackTrace = std::make_unique<Trace>(
+            QString("JS " + key.item + " callback").toStdString().c_str());
```

Comment:
why the unique_ptr? The trace object is not passed around, it could easily be allocated on the stack.
The creation of the QString also involves a heap allocation (likely since the string is likely quite long). `toStdString` might involves a heap allocation or at least a copy.

Reply:
The trace Object id designed to live on the stack and will have almost zero overhead in normal mode. 
You must not use QString here, because is also allocates memory in non developer mode. 

This is the correct usage for minimum over-head at non developer mode: 
https://github.com/mixxxdj/mixxx/blob/d5cf277e73ea96f68c366ce418431f18a4598406/src/soundio/sounddevicenetwork.cpp#L415

To speed up the debug case as well, you may consider to introduce a char16_t* overload of Trace and pass the string as u"  " 
Or even better a template taking char16_t[] to make use of sizeoff() instead of parsing for "/0"